frontend/src/app/main/data/workspace.cljs

```diff
@@ -1651,7 +1653,7 @@
 
                   all-objects (merge (:objects page) paste-objects)
 
-                  changes  (-> (dws/prepare-duplicate-changes all-objects page selected delta it nil)
+                  changes  (-> (dws/prepare-duplicate-changes all-objects page selected delta it libraries state)
```

Comment:
Passing the whole application state to a function introduces a huge implicit dependency. I think it would be better to pass only the part that is needed.